### PR TITLE
Do not require plugins to set 'since' field

### PR DESCRIPTION
**Pull request by azz**
Fixes the issue raised in https://github.com/prettier/plugin-php/issues/55

### CODE REVIEW

File: src/main/options.js [hunk 1]
```diff
@@ -81,10 +81,9 @@ function inferParser(filepath, plugins) {
     pluginsLoaded: true
   }).languages.find(
     language =>
-      typeof language.since === "string" &&
```
Comment (ikatyang) [inline src/main/options.js]: `language.since !== null`? We should omit unreleased languages here.

File: src/common/support.js [hunk 2]
```diff
@@ -255,7 +255,15 @@ function getSupportInfo(version, opts) {
...
-    .filter(language => language.since && semver.gte(version, language.since))
+    .filter(language => {
+      if (language.since === null) {
+        return false;
+      }
+      if (language.since) {
+        return semver.gte(version, language.since);
+      }
+      return true;
+    })
```
Comment (ikatyang) [inline src/common/support.js]: How about

```js
language.since
  ? semver.gte(version, language.since)
  : language.since !== null
```

?

Comment (azz) [reply src/common/support.js]: Yeah I originally had a ternary but it looked messy/hard to reason about so I switched it to if/else.

File: src/main/options.js (hunk 1 above)
Comment (azz) [reply src/main/options.js]: Good catch.

**Pull request merged**